packages/slider/src/SliderHandle.ts

```diff
@@ -0,0 +1,170 @@
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import { property, PropertyValues } from '@spectrum-web-components/base';
+import { Focusable } from '@spectrum-web-components/shared/src/focusable.js';
+import {
+    NumberFormatter,
+    NumberFormatOptions,
+} from '@internationalized/number';
+
+export type HandleMin = number | 'previous';
+export type HandleMax = number | 'next';
+
+export type HandleValues = {
+    name: string;
+    value: number;
+}[];
+
+export interface Controller {
+    inputForHandle(handle: SliderHandle): HTMLInputElement | undefined;
+    requestUpdate(): void;
+    setValueFromHandle(handle: SliderHandle): void;
+    language: string;
+}
+
+export type SliderNormalization = {
+    toNormalized: (value: number, min: number, max: number) => number;
+    fromNormalized: (value: number, min: number, max: number) => number;
+};
+
+export const defaultNormalization: SliderNormalization = {
+    toNormalized(value: number, min: number, max: number) {
+        return (value - min) / (max - min);
+    },
+    fromNormalized(value: number, min: number, max: number) {
+        return value * (max - min) + min;
+    },
+};
+
+const MinConverter = {
+    fromAttribute: (value: string): number | 'previous' => {
+        if (value === 'previous') return value;
+        return parseFloat(value);
+    },
+    toAttribute: (value: 'previous' | number): string => {
+        return value.toString();
+    },
+};
+
+const MaxConverter = {
+    fromAttribute: (value: string): number | 'next' => {
+        if (value === 'next') return value;
+        return parseFloat(value);
+    },
+    toAttribute: (value: 'next' | number): string => {
+        return value.toString();
+    },
+};
+
+export class SliderHandle extends Focusable {
+    public handleController?: Controller;
+
+    public get handleName(): string {
+        return this.name;
+    }
+
+    public get focusElement(): HTMLElement {
+        return this.handleController?.inputForHandle(this) ?? this;
+    }
+
+    @property({ type: Number })
+    value = 10;
+
+    @property({ type: Boolean, reflect: true })
+    public dragging = false;
+
+    @property({ type: Boolean })
+    public highlight = false;
+
+    @property({ type: String })
+    public name = '';
+
+    @property({ reflect: true, converter: MinConverter })
+    public min?: number | 'previous';
+
+    @property({ reflect: true, converter: MaxConverter })
+    public max?: number | 'next';
+
+    @property({ type: Number, reflect: true })
+    public step?: number;
+
+    @property({ type: Object, attribute: 'format-options' })
+    public formatOptions?: NumberFormatOptions;
+
+    @property({ type: String, attribute: 'aria-label' })
+    public ariaLabel?: string;
```

Comment:
This need to surface as the `label` attribute as any `aria-*` attribute will feed directly into the accessibility tree. And, while we want that on the `<input>` element, we don't want that on the `sp-slider-handle` element.

Reply:
It would be great to use the attribute in the stories and documentation, too.

Reply:
Any suggestions here? We could have a `label` attribute on `sp-slider-handle` that is passed through to `aria-label` on the `input`. It's a bit confusing to the developer using the component.

I did have one example in the docs [here](https://github.com/adobe/spectrum-web-components/blob/cuberoot/multi-handle-slider/packages/slider/slider-handle.md#L35), but more should be added.

Reply:
`<my-el label="Text">` is generally what we do here across the library. 

Can you expand on the confusion? I'm not sure how it's a different amount of confusion that placing `aria-label` on an element and then it being applied to two different things in the accessibility tree.

![Screen reader reporting of the Maximum Handle in the demo](https://user-images.githubusercontent.com/1156657/120705603-072d4080-c486-11eb-9883-a31cf1ca9058.png)

Beyond the attribute naming, this seems to interact with the screen reader quite nicely! If we can at least have  both versions here: https://cuberoot-multi-handle-slider--spectrum-web-components.netlify.app/components/slider-handle leverage this attribute, I think we'll be good for iteration 1.

Reply:
Ok, I will make it so. The confusion I meant was setting a `label` attribute that would never be rendered (only for aria). However, I think that is the best solution